cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -60,18 +60,20 @@ public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredAppli
         DiffLogic diffLogic = new DiffLogic();
         Log.info("Comparing the applications...");
         DiffResult wrappedDiff = diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
+        System.out.println(diffLogic.createDiffOutput(liveApplicationsConfig, desiredApplicationsConfig));
         Log.info("Applications compared.");
 
         Map<String, List<CfChange>> allApplicationChanges = wrappedDiff.getApplicationChanges();
+        ApplicationsOperations appOperations = new ApplicationsOperations(cfOperations);
 
-        Log.info("Applying changes to applications...");
-        for (Entry<String, List<CfChange>> applicationChangesEntry : allApplicationChanges.entrySet()) {
-            String applicationName = applicationChangesEntry.getKey();
-            Log.debug("Start applying the changes to the app:", applicationName);
-            List<CfChange> applicationChanges = applicationChangesEntry.getValue();
+        Flux.fromIterable(allApplicationChanges.entrySet())
+                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.create(appOperations, appChangeEntry.getKey(),
+                        appChangeEntry.getValue()))
+                .doOnSubscribe(subscription -> System.out.println(Thread.currentThread().getName()))
+                .onErrorContinue((throwable, o) -> throwable.printStackTrace())
+                .blockLast();
```

Comment:
To improve readability:
```suggestion
        Flux<Void> applicationRequests = Flux.fromIterable(allApplicationChanges.entrySet())
                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.create(appOperations, appChangeEntry.getKey(),
                        appChangeEntry.getValue()))
                .doOnSubscribe(subscription -> System.out.println(Thread.currentThread().getName()))
                .onErrorContinue((throwable, o) -> throwable.printStackTrace());
        applicationRequests.blockLast();
```